### PR TITLE
Add fuzz tests for `StorageVec::binary_search*`

### PR DESCRIPTION
**Pull request by cmichi**
Follow-up for https://github.com/paritytech/ink/pull/836.

### CODE REVIEW

File: crates/storage/src/collections/vec/fuzz_tests.rs [hunk 1]
```diff
@@ -83,3 +83,105 @@ crate::fuzz_storage!("vec_1", StorageVec<u32>);
...
+}
+
+#[quickcheck]
+fn fuzz_binary_search_nonexistent(std_vec: Vec<i32>) {
+    // given
+    if std_vec.len() == 0 {
+        return
+    }
+    let mut unique_std_vec: Vec<i32> = std_vec.into_iter().unique().collect();
+    let removed_el = unique_std_vec.pop().expect("first element must exist");
```
Comment (HCastano) [inline crates/storage/src/collections/vec/fuzz_tests.rs:122]:
```suggestion
    let removed_element = unique_std_vec.pop().expect("Length is non-zero, therefore at least one element must exist.");
```

File: crates/storage/src/collections/vec/fuzz_tests.rs [hunk 2]
```diff
@@ -83,3 +83,105 @@ crate::fuzz_storage!("vec_1", StorageVec<u32>);
...
+    let original_std_vec = std_vec.clone();
+    std_vec.sort();
+    let ink_vec = StorageVec::from_iter(std_vec.clone());
+
+    for x in original_std_vec {
+        // when
+        let index = std_vec
+            .binary_search(&x)
+            .expect("`x` must be found in `Vec`") as u32;
+        let index_sv = ink_vec
```
Comment (HCastano) [inline crates/storage/src/collections/vec/fuzz_tests.rs:102]:
```suggestion
        let ink_index = ink_vec
```

Maybe change this for consistency with the other tests?

File: crates/storage/src/collections/vec/fuzz_tests.rs [hunk 3]
```diff
@@ -83,3 +83,105 @@ crate::fuzz_storage!("vec_1", StorageVec<u32>);
...
+            .binary_search(&x)
+            .expect("`x` must be found in `StorageVec`");
+        let index_sv_by = ink_vec
```
Comment (HCastano) [inline crates/storage/src/collections/vec/fuzz_tests.rs:105]:
```suggestion
        let ink_index_by = ink_vec
```

Same here

File: crates/storage/src/collections/vec/fuzz_tests.rs [hunk 4]
```diff
@@ -83,3 +83,105 @@ crate::fuzz_storage!("vec_1", StorageVec<u32>);
...
+}
+#[quickcheck]
+fn fuzz_binary_search_by_key_nonexistent(std_vec: Vec<(i32, i32)>) {
+    // given
+    if std_vec.len() == 0 {
+        return
+    }
+    let mut unique_std_vec: Vec<(i32, i32)> =
+        std_vec.into_iter().unique_by(|&(_a, b)| b).collect();
+    let removed_el = unique_std_vec.pop().expect("first element must exist");
```
Comment (HCastano) [inline crates/storage/src/collections/vec/fuzz_tests.rs:173]: Same suggestions on naming here

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -415,7 +415,7 @@ fuzz-tests:
   <<:                              *test-refs
   variables:
     # The QUICKCHECK_TESTS default is 100
-    QUICKCHECK_TESTS:              20000
+    QUICKCHECK_TESTS:              5000
```
Comment (HCastano) [inline .gitlab-ci.yml:418]: Why is this being reduced? What kind of time savings are we looking at?

**Review by HCastano — APPROVED** [review]

File: .gitlab-ci.yml (hunk 5 above)
Comment (cmichi) [reply .gitlab-ci.yml:418]: See the extended commit message. Current duration is ~15 minutes for this stage, I'd like to see how this reduces with this lower number.

**Pull request merged**